### PR TITLE
Interpolation renderer: Adding another custom-renderer for openlayers

### PR DESCRIPTION
**Pull request by MichaelLangbein**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/dlr-eoc/ukis-frontend-libraries/blob/master/CONTRIBUTING.md#-commit-message-guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] The Project is [building and passes all tests](https://github.com/dlr-eoc/ukis-frontend-libraries/blob/master/DEVELOPMENT.md#further-you-can-test-and-build-locally) without errors

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?
We already have a selection of custom renderers for openlayers.  This PR extends that set with another layer.

## What is the new behavior?
With this PR another renderer is added: `InterpolationRenderer`.
   - Takes a vector-source (or a clustered vector-source) and does inverse-distance-interpolation on every pixel between the data points.
   - Allows user to optionally style data-point-labels just like any other openlayers vector-layer.
   - Additionally, the underlying WebGL library has been extended to better handle pure-data-textures (ie. textures that are not intended for display, but for data-transmission).

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information
This PR only affects the packages utils-maps and demo-maps (the latter to a lesser degree: the old custom renderers have merely been moved from demo-maps to utils-maps).

### CODE REVIEW

File: projects/utils-browser/src/lib/layout/colors.ts [hunk 1]
```diff
@@ -0,0 +1,177 @@
+
```
Comment (boeckMt) [inline projects/utils-browser/src/lib/layout/colors.ts:1]: For this we could use a small library e.g. https://gka.github.io/chroma.js so we don't have to reimplement all the stuff.

File: projects/utils-maps/src/lib/ol/customRenderers/interpolation_renderer.ts [hunk 2]
```diff
@@ -0,0 +1,600 @@
...
+    private interpolatedValues: Uint8Array;
+
+    private projection: Projection;
+    private bbox: number[];
+
+    constructor(layer: VectorLayer, private settings: InterpolationRendererSettings) {
+        super(layer);
+
+        // setting up HTML element
+        this.container = document.createElement('div');
```
Comment (boeckMt) [inline projects/utils-maps/src/lib/ol/customRenderers/interpolation_renderer.ts:108]: When interacting with the browser DOM the `angular document`  https://angular.io/api/common/DOCUMENT should be used.
```ts
import { Inject } from "@angular/core";
import { DOCUMENT } from '@angular/common'
...
constructor(@Inject(DOCUMENT) private document: Document)
...
```

**Review by boeckMt — CHANGES_REQUESTED** [review]
There are a loot of Stuff e.g. webgl/math.ts, webgl/engine.shapes.ts ... do we have to implement this or is there not any library to use for this?

Normally I'm not a fan of using to much dependency  but we also dont should try to reimplement the wheel ;)

**Comment by MichaelLangbein** [comment]
> For this we could use a small library e.g. https://gka.github.io/chroma.js so we don't have to reimplement all the stuff.
> There are a loot of Stuff e.g. webgl/math.ts, webgl/engine.shapes.ts ... do we have to implement this or is there not any library to use for this?

Sure thing, I'll find suitable replacements!
Chroma is of course nice. As for math, [glmatrix](http://glmatrix.net/) is the default when it comes to linalg. Update comming up soon!

**Comment by MichaelLangbein** [comment]
> > For this we could use a small library e.g. https://gka.github.io/chroma.js so we don't have to reimplement all the stuff.
> > There are a loot of Stuff e.g. webgl/math.ts, webgl/engine.shapes.ts ... do we have to implement this or is there not any library to use for this?
> 
> Sure thing, I'll find suitable replacements!
> Chroma is of course nice. As for math, [glmatrix](http://glmatrix.net/) is the default when it comes to linalg. Update comming up soon!

Alright, removed all non-essential math/color/shape/helpers stuff. 
Also: did not need to install chroma or glMatrix, because these are required on the *user*-side, meaning whoever installs `@dlr-eoc/utils-maps`.

File: projects/demo-maps/src/app/route-components/route-example-custom-layers/route-map4.component.ts [hunk 3]
```diff
@@ -49,7 +52,6 @@ export class RouteMap4Component implements OnInit, AfterViewInit {
 
 
   ngOnInit(): void {
-    this.mapSvc.setProjection('EPSG:4326');
```
Comment (boeckMt) [inline projects/demo-maps/src/app/route-components/route-example-custom-layers/route-map4.component.ts:52]: nice that you removed this, i noticed it is causing a problem with the Vector Tile layer 👍

Comment (MichaelLangbein) [reply projects/demo-maps/src/app/route-components/route-example-custom-layers/route-map4.component.ts:52]: Oh! Sorry, I had not noticed that. In that case, yeah, good riddance :)

File: projects/utils-browser/src/public-api.ts [hunk 4]
```diff
@@ -4,4 +4,3 @@
 
 export * from './lib/download/download';
 export * from './lib/layout/layout';
-export * from './lib/layout/colors';
```
Comment (boeckMt) [inline projects/utils-browser/src/public-api.ts]: How could you remove this file without including a library like chroma for it afterwards becasue I don't see any change in the package.json, or did you don't use the functions before?

Comment (MichaelLangbein) [reply projects/utils-browser/src/public-api.ts]: Those functions were only used on the client side - in another program depending on the ukis-frontend-libraries. 
That program has now been updated to use chroma instead.

**Review by boeckMt — APPROVED** [review]

**Review by boeckMt — APPROVED** [review]

**Pull request merged**